modelling/src/main/java/org/axonframework/modelling/command/inspection/AggregateModel.java

```diff
@@ -53,12 +54,32 @@
     }
 
     /**
-     * Gets the declared type based on given {@code type}.
+     * Gets the declared aggregate type based on given class {@code type}.
      *
      * @param type the type of the aggregate
      * @return the declared aggregate type represented by {@link String}, if exists
      */
     default Optional<String> declaredType(Class<?> type) {
         return Optional.empty();
     }
+
+    /**
+     * Gets a default constructor for a given class {@code type}.
+     *
+     * @param type the type of the aggregate
+     * @return the default constructor of this {@code type}, if exists
+     */
+    default Optional<Constructor<?>> defaultConstructor(Class<?> type) {
```

Comment:
As stated in another comment, I don't think the model should expose constructors. It forces implementation to use default constructors, instead of allowing AggregateFactory implementations to creates instances otherwise (e.g. using prototype Spring beans)